datafusion/common/src/scalar.rs

```diff
@@ -934,16 +928,20 @@ impl ScalarValue {
                         match values {
                             Some(values) => {
                                 // Push value for each field
-                                for c in 0..columns.len() {
-                                    let column = columns.get_mut(c).unwrap();
-                                    column.push(values[c].clone());
+                                for (i, v) in
```

Comment:
👍  this is a nicer implementation

Reply:
clippy complained on that :)

Reply:
All the better that removing a layer of `Box`ing lets clippy fix things more easily 👍

Reply:
A perhaps more idiomatic way to write this would be something like

```
for (column, value) in columns.iter_mut().zip(values)
```

Reply:
Yes, thats more concise. Fixed in 2 places.